src/components/PageDefinitions.js

```diff
@@ -256,11 +256,33 @@ class PageDefinitions extends AbstractPageDefinitions {
     this.props.dispatch(uiNotificationNew({ type: 'info', message, timeout: 5000 }))
   }
 
-  onDrop(acceptedFiles, rejectedFiles) {
+  onDragOver = e => e.preventDefault()
+  onDragEnter = e => e.preventDefault()
+
+  onDrop = e => {
+    e.preventDefault()
+    const text = e.dataTransfer.getData('Text')
+
+    if (text) {
+      this.onTextDrop(text)
+    } else {
+      const files = Object.values(e.dataTransfer.files)
+      const acceptedFiles = files[0].type === 'application/json'
```

Comment:
What about if files are multiple and some of them are not in the accepted format?
Another thing: `=== 'application/json'` couldn't be the only check that we will need to do.

I suggest to implement some methods in a utils/ class in order to pass dropped elements (files or texts) and do any check inside it.

Files checks could be multiple and not only `application/json`, and if a list of files contains one ore more rejected extensions , it shouldn't fail entirely, but otherwise parse only the accepted files (this is something is already happening i guess, check the codebase)